algolia/algoliasearch-sfcc-b2c
# feat: Content Indexing POC

## Content Indexing Proof of Concept (PoC)

### Objective
Develop a PoC for robust and efficient content indexing within our platform.

### Summary

- **Implementation**: Established a content indexing job, site preferences, and the core logic for indexing.
- **User Interface Integration**: Enhanced the Business Manager (BM) UI with new attributes to support indexing features.

### Features

- **Indexing Logic**: Adapted from product indexing, incorporating features like index type, additional attributes, threshold settings, monitoring, etc.
- **Content Division**: Enabled content splitting using a specific HTML tag, adjustable via job parameters.
- **Optimization for Size**: Implemented a mechanism to split records, capping at 10KB per record.
- **Search Engine Utilization**: Leveraged the native Salesforce Commerce Cloud (SFCC) search engine to identify all searchable components.
- **Comprehensive Indexing**: Included both Page Designer pages and Content Assets in the indexing process.

### Areas for Improvement & Next Steps

- **Modularity**: Enhance code reuse and efficiency between various indexing logic components.
- **Page Designer Content Indexing**: Refine the indexing of Page Designer pages to include searchable content.
- **Tag Management**: Introduce functionality to remove specific tags through job parameters (e.g., `<button>Click Here</button>`).
- **Testing**: Implement comprehensive test cases to validate all functionalities.
- **Record Size Handling**: Improve the record size management by reducing the current 250 Byte security margin through refined calculations.
- **Partial indexes & Dynamic Record Size**: Need to reconsider how we can update partial indexes (because content modifications also may change record splits) Probably, we need to remove this strategy for content indexing.

### Testing Instructions

1. **Job Import**: Import the new indexing job into the system.
2. **Execution**: Run the job and verify its execution.
3. **Verification**: Check for the visibility of all content assets on the Algolia dashboard.
4. **Splitter Tag Testing**: Test the splitting logic by altering the splitter tag in the job parameters.
5. **Attribute Testing**: To test extra attribute logic, add attributes in `AlgoliaContentConfig.js`.
6. **Content Checks**: Ensure only online and searchable content is indexed.
7. **Page Designer Integration**: For making Page Designer pages and components searchable, additional steps are required. Refer to the [RFC-3 - Content Indexing for SFCC](https://algolia.atlassian.net/wiki/spaces/SFCC/pages/4796745262/RFC-3+-+Content+Indexing+for+Salesforce+Commerce+Cloud+SFCC#Proposed-Approach-for-Indexing-Page-Designer-Content).
